linera-core/src/client/mod.rs

```diff
@@ -1243,29 +1301,46 @@ impl<Env: Environment> Client<Env> {
     }
 
     /// Downloads any certificates from the specified validator that we are missing for the given
-    /// chain, and processes them.
+    /// chain.
+    ///
+    /// If the chain is not in follow-only mode, also fetches and processes manager values
+    /// (timeout certificates, proposals, locking blocks) for consensus participation.
     #[instrument(level = "trace", skip(self, remote_node, chain_id))]
     pub(crate) async fn synchronize_chain_state_from(
         &self,
         remote_node: &RemoteNode<Env::ValidatorNode>,
         chain_id: ChainId,
     ) -> Result<(), chain_client::Error> {
-        let mut local_info = self.local_node.chain_info(chain_id).await?;
-        let query = ChainInfoQuery::new(chain_id).with_manager_values();
+        let with_manager_values = !self.is_chain_follow_only(chain_id);
+        let query = if with_manager_values {
+            ChainInfoQuery::new(chain_id).with_manager_values()
+        } else {
+            ChainInfoQuery::new(chain_id)
+        };
         let remote_info = remote_node.handle_chain_info_query(query).await?;
-        if let Some(new_info) = self
+        let local_info = self
             .download_certificates_from(remote_node, chain_id, remote_info.next_block_height)
-            .await?
-        {
-            local_info = new_info;
-        };
+            .await?;
+
+        if !with_manager_values {
+            return Ok(());
+        }
```

Comment:
For the case when the chain is in the follow-only mode, we return early here. But we didn't do any meaningful work before it – do we even eed to handle those two cases? What does the code above really do if the client is in the follow-only mode?

Reply:
It downloaded and processed all confirmed blocks. (We really need a better name for `download_certificates_from`.)

Reply:
OK, I forgot – I thought it's only _downloading_ (like the func name suggests).